{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# unified_abcc_chicago_epi_plot.R\n",
    "# Single script to plot ABCC/CHiC interactions together with epigenetic tracks\n",
    "# Requires: data.table, Sushi\n",
    "# Assumes input files similar to user's examples. Adjust paths and object names as needed.\n",
    "\n",
    "library(data.table)\n",
    "library(Sushi)    # for plotBed, plotBedpe, plotGenes, labelgenome\n",
    "\n",
    "library(ggplot2)\n",
    "library(GenomicRanges)\n",
    "\n",
    "library(biomaRt)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "atac_file      <- \"/rds/general/user/malyshev/home/miniPCHiC/hILCs/spivakov_pchic_ILC_CD4/data/peaks/ATAC/ILC3_ATAC_peaks.bed\"        # replace with your file/track\n",
    "k27ac_file     <- \"/rds/general/user/malyshev/home/miniPCHiC/hILCs/spivakov_pchic_ILC_CD4/data/peaks/CHIP/ILC3_H3K27ac_peaks.bed\"\n",
    "k4me3_file     <- \"/rds/general/user/malyshev/home/miniPCHiC/hILCs/spivakov_pchic_ILC_CD4/data/peaks/CHIP/ILC3_H3K4me3_peaks.bed\"\n",
    "\n",
    "epiread <- function(epifile){\n",
    "    epiprofread <- fread(epifile)\n",
    "    setnames(epiprofread, colnames(epiprofread[,1:3]), c(\"chr\", \"peak_start\", \"peak_end\"))\n",
    "    epiprofread[,chr := gsub(\"chr\", \"\", chr)]\n",
    "    return(epiprofread)\n",
    "}\n",
    "\n",
    "k27ac <- epiread(k27ac_file)\n",
    "atac <- epiread(atac_file)\n",
    "k4me3 <- epiread(k4me3_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "## plotting gene tracks\n",
    "\n",
    "gene_track_plot <- function(gene_name = gene_name, chrom = chrom, bmap = bmap, bigStart = bigStart, bigEnd = bigEnd, midpoint = midpoint){\n",
    "    \n",
    "    mart <- useEnsembl(\n",
    "    biomart = \"ensembl\", \n",
    "    dataset = \"hsapiens_gene_ensembl\"      \n",
    "    )\n",
    "\n",
    "    gene_annotation <- getBM(\n",
    "      attributes = c(\"chromosome_name\", \"exon_chrom_start\", \"exon_chrom_end\", \"external_gene_name\", \"strand\", \"gene_biotype\"),\n",
    "      filters = c(\"chromosome_name\", \"start\", \"end\"),\n",
    "      values = list(chrom, bigStart*1000, bigEnd*1000),\n",
    "      mart = mart\n",
    "    )\n",
    "\n",
    "    gene_annotation <- as.data.table(gene_annotation)\n",
    "\n",
    "    gene_annotation[, `:=`(\n",
    "      chromosome_name = as.character(chromosome_name),\n",
    "      strand = as.character(strand)\n",
    "    )]\n",
    "    \n",
    "    # keep only protein-coding genes\n",
    "    gene_annotation <- gene_annotation[gene_biotype == \"protein_coding\"]\n",
    "\n",
    "    gene_annotation <- gene_annotation[, .(\n",
    "      chrom = chromosome_name, \n",
    "      start = exon_chrom_start, \n",
    "      stop = exon_chrom_end, \n",
    "      gene = external_gene_name, \n",
    "      strand, \n",
    "      colour = adjustcolor(\"grey23\", alpha.f = 0.5)\n",
    "      #colour = ifelse(strand == 1, \"darkgoldenrod\", \"dodgerblue4\")\n",
    "      )]\n",
    "\n",
    "    for(current_gannot in gene_annotation[, sort(unique(gene))]){\n",
    "          if(grepl(current_gannot, gene_name) == TRUE){\n",
    "            gene_annotation[(gene == current_gannot), colour := \"dodgerblue4\"]\n",
    "          }\n",
    "        }\n",
    "\n",
    "    gene_annotation[(strand == 1 | strand == -1), strand := ifelse(strand == 1, \"+\", \"-\")]\n",
    "    gene_annotation <- gene_annotation[gene!=\"\"]\n",
    "        \n",
    "    return(gene_annotation)\n",
    "\n",
    "}\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(patchwork)\n",
    "\n",
    "batch_epi_plot_short <- function(peakm, imputed, bait_list, bmap, rmap,\n",
    "                                 window_left = NULL, window_right = NULL, zoom = TRUE) {\n",
    "  \n",
    "  for (id in bait_list) {\n",
    "    message(\"Processing bait: \", id)\n",
    "    gene_name <- bmap[V4 == id]$V5[1]\n",
    "    \n",
    "    # ----------------- Zoom/window definition -----------------\n",
    "    if (zoom) {\n",
    "      if (is.null(window_left)) {\n",
    "        window_left <- (min(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist))) / 1000 - 20\n",
    "      }\n",
    "      if (is.null(window_right)) {\n",
    "        window_right <- (max(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist))) / 1000 + 20\n",
    "      }\n",
    "    } else {\n",
    "      if (is.null(window_left)) {\n",
    "        window_left <- (min(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist))) / 1000 - 20\n",
    "      }\n",
    "      if (is.null(window_right)) {\n",
    "        window_right <- (max(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist))) / 1000 + 20\n",
    "      }\n",
    "    }\n",
    "    \n",
    "    midpoint <- (bmap[V4==id]$V2 + bmap[V4==id]$V3) / 2000  # kb scale\n",
    "    bigStart <- midpoint - abs(window_left)\n",
    "    bigEnd   <- midpoint + abs(window_left)\n",
    "    chrom    <- bmap[V4==id][1,]$V1\n",
    "    \n",
    "    peakm_sel <- peakm[baitChr == chrom][((bait_start_fres > bigStart*1000) & (oe_end_fres < bigEnd*1000)) | \n",
    "                                           ((oe_start_fres > bigStart*1000) & (bait_end_fres < bigEnd*1000))]\n",
    "    \n",
    "    # ----------------- Output file -----------------\n",
    "    fn <- paste0(\"~/miniPCHiC/hILCs/hILCs_run2/abc_results/\", \n",
    "                 id, \"_\", window_left, \"kb_left_\", window_right, \"kb_right_zoom_\", zoom, \"_5kb.pdf\")\n",
    "    \n",
    "    pdf(fn, width = 10, height = 10)\n",
    "    \n",
    "    # ----------------- Contact plot -----------------\n",
    "    this <- imputed[baitID == id]\n",
    "    \n",
    "    this_abcc    <- unique(xpm[baitID_5kb == id & !is.na(ABC.Score)], by = c(\"baitID_5kb\", \"oeID_5kb\"))\n",
    "    this_chicago <- unique(xpm[baitID_5kb == id & !is.na(chicago_score_5kb)], by = c(\"baitID_5kb\", \"oeID_5kb\"))\n",
    "    this_both    <- merge(this_abcc, this_chicago, by = c(\"baitID_5kb\", \"oeID_5kb\"))\n",
    "    \n",
    "    this <- this %>%\n",
    "      mutate(group = case_when(\n",
    "        otherEndID %in% this_both$oeID_5kb ~ \"Both\",\n",
    "        otherEndID %in% this_abcc$oeID_5kb ~ \"ABCC\",\n",
    "        otherEndID %in% this_chicago$oeID_5kb ~ \"CHiCAGO\",\n",
    "        TRUE ~ \"None\"\n",
    "      ))\n",
    "    \n",
    "    color_map <- c(\"ABCC\" = \"blue\", \"CHiCAGO\" = \"red\", \"Both\" = \"green\", \"None\" = \"black\")\n",
    "    \n",
    "    rmap_abcc    <- rmap[V4 %in% this_abcc$oeID_5kb]\n",
    "    rmap_chicago <- rmap[V4 %in% this_chicago$oeID_5kb]\n",
    "    rmap_both    <- rmap[V4 %in% this_both$oeID_5kb]\n",
    "    \n",
    "    highlights <- bind_rows(\n",
    "      mutate(as.data.frame(rmap_abcc), group = \"ABCC\", col = \"blue\"),\n",
    "      mutate(as.data.frame(rmap_chicago), group = \"CHiCAGO\", col = \"red\"),\n",
    "      mutate(as.data.frame(rmap_both), group = \"Both\", col = \"green\")\n",
    "    )\n",
    "    \n",
    "    g_contacts <- ggplot(this, aes(x = (oe_start + oe_end)/2)) +\n",
    "          geom_line(aes(y = N_imp.x), linetype = \"dashed\") +\n",
    "          geom_point(\n",
    "            aes(\n",
    "              y = N_obs,\n",
    "              colour = group,                # border colour\n",
    "              fill   = ifelse(N_obs >= N_imp.x, group, NA)  # fill same as group if above expected\n",
    "            ),\n",
    "            shape = 21, size = 2.2, stroke = 0.7\n",
    "          ) +\n",
    "          scale_colour_manual(values = color_map, guide = \"none\") +\n",
    "          scale_fill_manual(values = color_map, na.value = \"white\", guide = \"none\") +\n",
    "          coord_cartesian(xlim = c(bigStart * 1000, bigEnd * 1000)) +\n",
    "          theme_minimal(base_size = 10) +\n",
    "          theme(\n",
    "            panel.grid = element_blank(),\n",
    "            panel.border = element_blank(),\n",
    "            axis.text = element_text(size = 10),\n",
    "              axis.line    = element_line(colour = \"black\")\n",
    "          ) +\n",
    "          labs(y = \"Contact\", x = NULL) +\n",
    "          geom_rect(\n",
    "            data = highlights,\n",
    "            aes(xmin = V2, xmax = V3),\n",
    "            fill = highlights$col,\n",
    "            ymin = -Inf, ymax = Inf, alpha = 0.1, inherit.aes = FALSE\n",
    "          )\n",
    "\n",
    "    \n",
    "    # ----------------- Gene track (your current version) -----------------\n",
    "\n",
    "    gene_sel <- gene_track_plot(gene_name = gene_name, chrom = chrom, \n",
    "                                  bmap = bmap, bigStart = bigStart, bigEnd = bigEnd, midpoint = midpoint)\n",
    "    gene_sel <- gene_sel %>%\n",
    "      mutate(row = as.numeric(factor(gene)))  \n",
    "    \n",
    "    gene_plot <- gene_sel %>%\n",
    "      group_by(gene, row, strand) %>%\n",
    "      summarise(start_gene = min(start), stop_gene = max(stop), .groups = \"drop\") %>%\n",
    "      mutate(label_x = ifelse(strand == \"+\", start_gene - 0.05*(bigEnd-bigStart)*1000,\n",
    "                              stop_gene + 0.05*(bigEnd-bigStart)*1000),\n",
    "             hjust_val = ifelse(strand == \"+\", 0, 1),\n",
    "             highlight = (gene == gene_name))\n",
    "    \n",
    "    gene_sel <- gene_sel %>%\n",
    "      mutate(highlight = (gene == gene_name))\n",
    "    \n",
    "    g_genes <- ggplot() +\n",
    "      geom_segment(data = gene_plot,\n",
    "                   aes(x = start_gene, xend = stop_gene, y = row, yend = row),\n",
    "                   size = 0.1, color = \"gray70\") +\n",
    "      geom_rect(data = gene_sel,\n",
    "                aes(xmin = start, xmax = stop, ymin = row - 0.2, ymax = row + 0.2,\n",
    "                    fill = highlight),\n",
    "                color = \"black\",size = 0.1) +\n",
    "      scale_fill_manual(values = c(\"TRUE\" = \"orange\", \"FALSE\" = \"gray80\"), guide=\"none\") +\n",
    "      geom_text(data = gene_plot,\n",
    "                aes(x = label_x, y = row, label = gene),\n",
    "                size = 3) +\n",
    "      scale_y_continuous(breaks = NULL) +\n",
    "      theme_minimal(base_size = 10) +\n",
    "      theme(panel.grid = element_blank())+\n",
    "      labs(y = \"Genes\", fill = \"\") +\n",
    "      coord_cartesian(xlim = c(bigStart*1000, bigEnd*1000))\n",
    "    \n",
    "    # ----------------- Epigenetic tracks -----------------\n",
    "    safe_df <- function(beddata) {\n",
    "      if (!is.null(beddata) && nrow(beddata) > 0) as.data.frame(beddata)\n",
    "      else data.frame(\"peak_start\" = numeric(0), \"peak_end\" = numeric(0))\n",
    "    }\n",
    "    \n",
    "    atac_df  <- safe_df(atac)\n",
    "    k27ac_df <- safe_df(k27ac)\n",
    "    k4me3_df <- safe_df(k4me3)\n",
    "    \n",
    "    plot_chip <- function(df, label) {\n",
    "      ggplot(df, aes(xmin = peak_start, xmax = peak_end, ymin = 0, ymax = 1)) +\n",
    "        geom_rect(fill = \"grey40\") +\n",
    "        labs(y = label, x = NULL) +\n",
    "        coord_cartesian(xlim = c(bigStart*1000, bigEnd*1000), ylim = c(0,1)) +\n",
    "        theme_minimal(base_size = 10) +\n",
    "        theme(panel.grid = element_blank(), \n",
    "              panel.border = element_blank(),\n",
    "              axis.text.x=element_blank(),\n",
    "              axis.ticks.x=element_blank(),\n",
    "             axis.text.y=element_blank(),\n",
    "             axis.ticks.y=element_blank())\n",
    "    }\n",
    "    \n",
    "    g_atac  <- plot_chip(atac_df,  \"ATAC\")\n",
    "    g_k27ac <- plot_chip(k27ac_df, \"H3K27ac\")\n",
    "    g_k4me3 <- plot_chip(k4me3_df, \"H3K4me3\")\n",
    "    \n",
    "    # ----------------- Final combined plot -----------------\n",
    "    final_plot <- (g_contacts / g_atac / g_k27ac / g_k4me3 / g_genes) +\n",
    "      plot_layout(heights = c(2,0.2, 0.2, 0.2, 0.5), guides = \"collect\")\n",
    "    \n",
    "    # ----------------- Save to PDF -----------------\n",
    "    print(final_plot)\n",
    "    dev.off()\n",
    "  }\n",
    "}\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "imputed = fread(\"~/miniPCHiC/hILCs/hILCs_run2/abc_results/imputed_abcc_chicago_shared_baits_sample100.txt\")\n",
    "xpm = fread(\"~/miniPCHiC/hILCs/ILC3/PCHiC/data/ILC3_chicago_fres_bin_5kb_abc_023_fres_extended_peakm_13012025.txt\")\n",
    "bmap <- fread(\"~/spivakov_analysis_live/Design/Human_hg38_bin5K_sol_baits/human_DpnII_5K_sol_baits.baitmap\")\n",
    "rmap <- fread(\"~/spivakov_analysis_live/Design/Human_hg38_bin5K_sol_baits/human_DpnII_5K_sol_baits.rmap\")\n",
    "\n",
    "bait_list <- c(253107, 287095, 297402, 372379, 421189, 425229)\n",
    "\n",
    "batch_epi_plot_short(peakm = xpm, imputed = imputed, \n",
    "                     bait_list = bait_list, bmap = bmap, \n",
    "                     rmap = rmap, window_left = -100, window_right = 100, zoom = FALSE)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:new-r-env]",
   "language": "R",
   "name": "conda-env-new-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
