{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“package ‘Chicago’ was built under R version 3.6.3”\n",
      "Warning message:\n",
      "“replacing previous import ‘vctrs::data_frame’ by ‘tibble::data_frame’ when loading ‘dplyr’”\n",
      "\n",
      "\n",
      "Welcome to CHiCAGO - version 1.14.0\n",
      "\n",
      "If you are new to CHiCAGO, please consider reading the vignette through the command: vignette(\"Chicago\").\n",
      "\n",
      "NOTE: Default values of tlb.minProxOEPerBin and tlb.minProxB2BPerBin changed as of Version 1.1.5. No action is required unless you specified non-default values, or wish to re-run the pipeline on old chicagoData objects. See news(package=\"Chicago\")\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomicRanges’ was built under R version 3.6.3”\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Warning message:\n",
      "“package ‘BiocGenerics’ was built under R version 3.6.3”\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Chicago’:\n",
      "\n",
      "    params\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘IRanges’ was built under R version 3.6.3”\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomeInfoDb’ was built under R version 3.6.3”\n",
      "Loading required package: zoo\n",
      "\n",
      "\n",
      "Attaching package: ‘zoo’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    as.Date, as.Date.numeric\n",
      "\n",
      "\n",
      "Loading required package: biomaRt\n",
      "\n",
      "Warning message:\n",
      "“package ‘biomaRt’ was built under R version 3.6.3”\n"
     ]
    }
   ],
   "source": [
    "### plotting chic_abc_epi sushi version\n",
    "### Fig.5 D,E and Fig. 6A\n",
    "\n",
    "library(data.table)\n",
    "library(Chicago)\n",
    "library(ggplot2)\n",
    "library(GenomicRanges)\n",
    "\n",
    "library(Sushi)\n",
    "library(biomaRt)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "### loading chicago/abc peakmatrix\n",
    "\n",
    "ilc_chic_abc_peakm <- fread(\"ILC3_chicago_fres_5kb_abc_fres_extended_peakm.txt\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "### loading epi files\n",
    "\n",
    "k27ac <- \"ILC3_H3K27ac_peaks.bed\"\n",
    "atac <- \"ILC3_ATAC_peaks.bed\"\n",
    "\n",
    "k36me3 <- \"GSE50893_SNYDER_HG38_GM12878_H3K36ME3_VS_SNYDER_HG19_GM12878_INPUT_reconcile.dedup_peaks.encodePeak.txt\"\n",
    "ikzf1 <- \"ikzf1_narrowpeak.bed\"\n",
    "batf <- \"batf_narrowpeak.bed\"\n",
    "\n",
    "nfkb<- \"nfkb_narrowpeak.bed\"\n",
    "atf2 <- \"atf2_narrowpeak.bed\"\n",
    "\n",
    "ctcf <- \"GSE50893_SNYDER_HG38_GM12878_CTCF_VS_SNYDER_HG19_GM12878_INPUT_reconcile.dedup_peaks.encodePeak.txt\"\n",
    "sa1 <- \"GSE50893_SNYDER_HG38_GM12878_SA1_VS_SNYDER_HG19_GM12878_INPUT_reconcile.dedup_peaks.encodePeak.txt\"\n",
    "\n",
    "epiread <- function(epifile){\n",
    "    epiprofread <- fread(epifile)\n",
    "    setnames(epiprofread, colnames(epiprofread[,1:3]), c(\"chr\", \"peak_start\", \"peak_end\"))\n",
    "    epiprofread[,chr := gsub(\"chr\", \"\", chr)]\n",
    "    return(epiprofread)\n",
    "}\n",
    "\n",
    "k27ac <- epiread(k27ac)\n",
    "atac <- epiread(atac)\n",
    "\n",
    "k36me3 <- epiread(k36me3)\n",
    "ikzf1 <- epiread(ikzf1)\n",
    "batf <- epiread(batf)\n",
    "\n",
    "nfkb <- epiread(nfkb)\n",
    "atf2 <- epiread(atf2)\n",
    "\n",
    "ctcf <- epiread(ctcf)\n",
    "sa1 <- epiread(sa1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "### loading baitmap file\n",
    "\n",
    "## fres\n",
    "bmap <- fread(\"hg38_dpnII.baitmap\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "### gwas data\n",
    "rmap_fres <- fread(\"hg38_dpnII.rmap\")\n",
    "rmap_5kb <- fread(\"human_DpnII_5K_sol_baits.rmap\")\n",
    "\n",
    "raw_deLange_data <- fread(\"28067908-GCST004132-EFO_0000384.h.tsv.gz\")\n",
    "ppi_deLange <- fread(\"CD_deLange_ILCs_hg38_ClassicCOGS_combinedInteractions_ALL/COGS_PPIs_data.table.txt\")\n",
    "ppi_deLange_susie <- fread(\"CD_deLange_ILCs_hg38_SuSIE_combinedInteractions_ALL/COGS_PPIs_data.table.txt\")\n",
    "\n",
    "delange_res <- fread(\"CD_deLange_ILCs_hg38_ClassicCOGS_combinedInteractions_ALL/Annotated_COGS_scores_data.table.txt\")\n",
    "delange_susie_res <- fread(\"CD_deLange_ILCs_hg38_SuSIE_combinedInteractions_ALL/Annotated_COGS_scores_data.table.txt\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "## merging gwas data with rmap\n",
    "\n",
    "ppirmap <- function(gwas, rmap_fres, rmap_5kb){\n",
    "  \n",
    "    gwas <- gwas[ppi >= 0.05]\n",
    "    setnames(gwas, colnames(gwas[,1]), \"V1\")\n",
    "\n",
    "    # overlap_by: \"summit\" or \"region\"\n",
    "  \n",
    "    # define middle point of a peak\n",
    "    gwas[,mid := pos]\n",
    "    gwas <- gwas[,c(\"V1\", \"mid\", \"pos\")]\n",
    "    gwas[,V1 := gsub(\"chr\", \"\", V1)]\n",
    "  \n",
    "    # assign peaks to fres rmap bins \n",
    "    setkey(gwas, V1, mid, pos)\n",
    "    setkey(rmap_fres, V1, V2, V3)\n",
    "    gwas_rmap <- foverlaps(gwas, rmap_fres, by.x = c(\"V1\", \"mid\", \"pos\"), by.y = c(\"V1\", \"V2\", \"V3\"))\n",
    "    gwas_rmap <- gwas_rmap[,c(\"V1\", \"mid\", \"pos\", \"V4\")]\n",
    "    setnames(gwas_rmap, \"V4\", \"fres_ID\")\n",
    "\n",
    "    # assign peaks to 5kb rmap bins by summit\n",
    "    setkey(gwas_rmap, V1, mid, pos)\n",
    "    setkey(rmap_5kb, V1, V2, V3)\n",
    "    gwas_rmap <- foverlaps(gwas_rmap, rmap_5kb, by.x = c(\"V1\", \"mid\", \"pos\"), by.y = c(\"V1\", \"V2\", \"V3\"))\n",
    "    gwas_rmap <- gwas_rmap[,c(\"V1\", \"mid\", \"pos\", \"fres_ID\", \"V4\")]\n",
    "    setnames(gwas_rmap, \"V4\", \"5kb_ID\")\n",
    "    \n",
    "    gwas_rmap[,mid := NULL]\n",
    "  \n",
    "  return(gwas_rmap)\n",
    "}\n",
    "\n",
    "# assigning gwas snps to rmaps\n",
    "\n",
    "ppi_deLange_rmap <- ppirmap(ppi_deLange, rmap_fres, rmap_5kb)\n",
    "ppi_deLange_susie_rmap <- ppirmap(ppi_deLange_susie, rmap_fres, rmap_5kb)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "## merging epigenetic data with rmap\n",
    "\n",
    "epirmap <- function(epiprof, rmap_fres, rmap_5kb, overlap_by){\n",
    "  \n",
    "  if (\"pos\" %in% colnames(epiprof)){\n",
    "    epiprof[,c(\"V2\", \"V3\") := NULL]\n",
    "    epiprof[,V2 := pos]\n",
    "    epiprof[,V3 := pos]\n",
    "  }else{\n",
    "    setnames(epiprof, colnames(epiprof[,1:3]), c(\"V1\", \"V2\", \"V3\"))\n",
    "  }    \n",
    "  \n",
    "  # overlap_by: \"summit\" or \"region\"\n",
    "  \n",
    "  if (overlap_by == \"summit\"){\n",
    "    \n",
    "      # define middle point of a peak\n",
    "      epiprof[,mid := (V2 + V3) /2]\n",
    "      epiprof[,mid_2 := (V2 + V3) /2]\n",
    "      epiprof <- epiprof[,c(\"V1\", \"V2\", \"V3\", \"mid\", \"mid_2\")]\n",
    "      epiprof[,V1 := gsub(\"chr\", \"\", V1)]\n",
    "  }else{\n",
    "      # define middle point of a peak\n",
    "      epiprof[,mid := V2]\n",
    "      epiprof[,mid_2 := V3]\n",
    "      epiprof <- epiprof[,c(\"V1\", \"V2\", \"V3\", \"mid\", \"mid_2\")]\n",
    "      epiprof[,V1 := gsub(\"chr\", \"\", V1)]\n",
    "  }\n",
    "\n",
    "  \n",
    "  # assign peaks to fres rmap bins by summit\n",
    "  setkey(epiprof, V1, mid, mid_2)\n",
    "  setkey(rmap_fres, V1, V2, V3)\n",
    "  epi_rmap <- foverlaps(epiprof, rmap_fres, by.x = c(\"V1\", \"mid\", \"mid_2\"), by.y = c(\"V1\", \"V2\", \"V3\"))\n",
    "  epi_rmap <- epi_rmap[,c(\"V1\", \"mid\", \"mid_2\", \"V4\")]\n",
    "  setnames(epi_rmap, \"V4\", \"fres_ID\")\n",
    "  \n",
    "  # assign peaks to 5kb rmap bins by summit\n",
    "  setkey(epi_rmap, V1, mid, mid_2)\n",
    "  setkey(rmap_5kb, V1, V2, V3)\n",
    "  epi_rmap <- foverlaps(epi_rmap, rmap_5kb, by.x = c(\"V1\", \"mid\", \"mid_2\"), by.y = c(\"V1\", \"V2\", \"V3\"))\n",
    "  epi_rmap <- epi_rmap[,c(\"V1\", \"mid\", \"mid_2\", \"fres_ID\", \"V4\")]\n",
    "  setnames(epi_rmap, \"V4\", \"5kb_ID\")\n",
    "\n",
    "  \n",
    "  return(epi_rmap)\n",
    "}\n",
    "\n",
    "# load rmaps\n",
    "rmap_fres <- fread(\"/rds/general/user/malyshev/home/spivakov_analysis_live/Design/Human_DpnII_binsize1500_maxL75K/hg38_dpnII.rmap\")\n",
    "rmap_5kb <- fread(\"/rds/general/user/malyshev/home/spivakov_analysis_live/Design/Human_hg38_bin5K_sol_baits/human_DpnII_5K_sol_baits.rmap\")\n",
    "\n",
    "atac_rmap <- epirmap(atac, rmap_fres, rmap_5kb, \"summit\")\n",
    "k27ac_rmap <- epirmap(k27ac, rmap_fres, rmap_5kb, \"summit\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "### prep mix resolution bedpe \n",
    "\n",
    "ilc_chic_abc_peakm_abc_only <- ilc_chic_abc_peakm[(is.na(chicago_fres) | chicago_fres <5) & (is.na(chicago_5kb) | chicago_5kb <5) &  ABC.Score >= 5]\n",
    "ilc_chic_abc_peakm_chic5kb <- ilc_chic_abc_peakm[chicago_5kb >= 5]\n",
    "ilc_chic_abc_peakm_chicfres <- ilc_chic_abc_peakm[(is.na(chicago_5kb) | chicago_5kb <5) & chicago_fres >= 5]\n",
    "\n",
    "## don't need to collapse ilc_chic_abc_peakm_abc_only, as interactions are already unique\n",
    "setnames(ilc_chic_abc_peakm_abc_only, \"ABC.Score\", \"score\")\n",
    "ilc_chic_abc_peakm_abc_only <- ilc_chic_abc_peakm_abc_only[,c(\"baitChr\", \"baitStart\", \"baitEnd\", \"oeChr\", \"oeStart\", \"oeEnd\", \"score\", \"baitID\", \"oeID\", \"baitName\")]\n",
    "ilc_chic_abc_peakm_abc_only[, res_type := \"ABC\"]\n",
    "ilc_chic_abc_peakm_abc_only[, delange_susie := ifelse(((baitID %in% ppi_deLange_susie_rmap$fres_ID) | (oeID %in% ppi_deLange_susie_rmap$fres_ID)), \"TRUE\", \"FALSE\")]\n",
    "ilc_chic_abc_peakm_abc_only[, atac := ifelse(oeID %in% atac_rmap$fres_ID, \"TRUE\", \"FALSE\")]\n",
    "ilc_chic_abc_peakm_abc_only[, k27ac := ifelse(oeID %in% k27ac_rmap$fres_ID, \"TRUE\", \"FALSE\")]\n",
    "\n",
    "\n",
    "## collapse 5kb interactions\n",
    "ilc_chic_abc_peakm_chic5kb_coll <- ilc_chic_abc_peakm_chic5kb[,list(baitChr = baitChr[1], baitStart = baitStart[1], baitEnd = baitStart[1],\n",
    "                                                                    oeChr = oeChr[1], oeStart = min(oeStart), oeEnd = max(oeEnd), score = chicago_5kb[1], baitName = baitName[1]), \n",
    "                                                              by = c(\"baitID_5kb\", \"oeID_5kb\")]\n",
    "ilc_chic_abc_peakm_chic5kb_coll <- ilc_chic_abc_peakm_chic5kb_coll[,c(\"baitChr\", \"baitStart\", \"baitEnd\", \"oeChr\", \"oeStart\", \"oeEnd\", \"score\", \"baitID_5kb\", \"oeID_5kb\", \"baitName\")]\n",
    "setnames(ilc_chic_abc_peakm_chic5kb_coll, c(\"baitID_5kb\", \"oeID_5kb\"), c(\"baitID\", \"oeID\"))\n",
    "ilc_chic_abc_peakm_chic5kb_coll[, res_type := \"chicago_5kb\"]\n",
    "ilc_chic_abc_peakm_chic5kb_coll[, delange_susie := ifelse(((baitID %in% ppi_deLange_susie_rmap$'5kb_ID') | (oeID %in% ppi_deLange_susie_rmap$'5kb_ID')), \"TRUE\", \"FALSE\")]\n",
    "ilc_chic_abc_peakm_chic5kb_coll[, atac := ifelse(oeID %in% atac_rmap$'5kb_ID', \"TRUE\", \"FALSE\")]\n",
    "ilc_chic_abc_peakm_chic5kb_coll[, k27ac := ifelse(oeID %in% k27ac_rmap$'5kb_ID', \"TRUE\", \"FALSE\")]\n",
    "               \n",
    "\n",
    "\n",
    "## don't need to collapse ilc_chic_abc_peakm_abc_only, as interactions are already unique\n",
    "setnames(ilc_chic_abc_peakm_chicfres, \"chicago_fres\", \"score\")\n",
    "ilc_chic_abc_peakm_chicfres <- ilc_chic_abc_peakm_chicfres[,c(\"baitChr\", \"baitStart\", \"baitEnd\", \"oeChr\", \"oeStart\", \"oeEnd\", \"score\", \"baitID\", \"oeID\", \"baitName\")]\n",
    "ilc_chic_abc_peakm_chicfres[, res_type := \"chicago_fres\"]\n",
    "ilc_chic_abc_peakm_chicfres[, delange_susie := ifelse(((baitID %in% ppi_deLange_susie_rmap$fres_ID) | (oeID %in% ppi_deLange_susie_rmap$fres_ID)), \"TRUE\", \"FALSE\")]\n",
    "ilc_chic_abc_peakm_chicfres[, atac := ifelse(oeID %in% atac_rmap$fres_ID, \"TRUE\", \"FALSE\")]\n",
    "ilc_chic_abc_peakm_chicfres[, k27ac := ifelse(oeID %in% k27ac_rmap$fres_ID, \"TRUE\", \"FALSE\")]\n",
    "   \n",
    "\n",
    "## merging\n",
    "peakm_remerge <- rbind(ilc_chic_abc_peakm_chicfres, ilc_chic_abc_peakm_chic5kb_coll, ilc_chic_abc_peakm_abc_only)\n",
    "setkey(peakm_remerge, baitChr, baitStart, baitEnd, oeChr, oeStart, oeEnd)\n",
    "\n",
    "## colouring interactions (colourblind safe): ABC - #1b9e77,green; chicago_5kb - #d95f02,red ; chicago_fres - #7570b3, purple\n",
    "peakm_remerge[,colour := ifelse(res_type == \"ABC\", \"#1b9e77\",\n",
    "                               ifelse(res_type == \"chicago_5kb\", \"#d95f02\", \"#7570b3\"))]\n",
    "\n",
    "peakm_remerge[, dist := ifelse(baitChr == oeChr, abs((oeEnd + oeStart)/2 - (baitStart + baitStart)/2), \"NA\")]\n",
    "\n",
    "fwrite(peakm_remerge, \"peakm_merged_chicago_fres_5kb_abc_fres_epiplot_lola.txt\", sep = \"\\t\", scipen = 999)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "## plotting gene tracks\n",
    "\n",
    "gene_track_plot <- function(gene_name = gene_name, chrom = chrom, bmap = bmap, bigStart = bigStart, bigEnd = bigEnd, midpoint = midpoint){\n",
    "    \n",
    "    mart <- useEnsembl(\n",
    "    biomart = \"ensembl\", \n",
    "    dataset = \"hsapiens_gene_ensembl\"      \n",
    "    )\n",
    "\n",
    "    gene_annotation <- getBM(\n",
    "      attributes = c(\"chromosome_name\", \"exon_chrom_start\", \"exon_chrom_end\", \"external_gene_name\", \"strand\"),\n",
    "      filters = c(\"chromosome_name\", \"start\", \"end\"),\n",
    "      values = list(chrom, bigStart*1000, bigEnd*1000),\n",
    "      mart = mart\n",
    "    )\n",
    "\n",
    "    gene_annotation <- as.data.table(gene_annotation)\n",
    "\n",
    "    gene_annotation[, `:=`(\n",
    "      chromosome_name = as.character(chromosome_name),\n",
    "      strand = as.character(strand)\n",
    "    )]\n",
    "\n",
    "    gene_annotation <- gene_annotation[, .(\n",
    "      chrom = chromosome_name, \n",
    "      start = exon_chrom_start, \n",
    "      stop = exon_chrom_end, \n",
    "      gene = external_gene_name, \n",
    "      strand, \n",
    "      colour = adjustcolor(\"grey23\", alpha.f = 0.5)\n",
    "      #colour = ifelse(strand == 1, \"darkgoldenrod\", \"dodgerblue4\")\n",
    "      )]\n",
    "\n",
    "    for(current_gannot in gene_annotation[, sort(unique(gene))]){\n",
    "          if(grepl(current_gannot, gene_name) == TRUE){\n",
    "            gene_annotation[(gene == current_gannot), colour := \"dodgerblue4\"]\n",
    "          }\n",
    "        }\n",
    "\n",
    "    gene_annotation[(strand == 1 | strand == -1), strand := ifelse(strand == 1, \"+\", \"-\")]\n",
    "    gene_annotation <- gene_annotation[gene!=\"\"]\n",
    "        \n",
    "    return(gene_annotation)\n",
    "\n",
    "}\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "batch_epi_plot_short <- function(peakm, bait_list, bmap, window_left = window_left, window_right = window_right, zoom = zoom, all_ints = all_ints, idonly = TRUE){\n",
    "\n",
    "    ## e.g. peakm can be ilc_chic_abc_peakm\n",
    "    for (id in bait_list){\n",
    "        \n",
    "    gene_name <- bmap[V4 == id]$V5[1]\n",
    "\n",
    "        if(zoom == TRUE){\n",
    "\n",
    "          if (is.null(window_left)){\n",
    "            window_left<- (min(as.numeric(peakm[baitID == id][baitChr == oeChr][delange_susie == TRUE]$dist)))/1000 - 20\n",
    "          }else{\n",
    "            window_left = window_left\n",
    "          }\n",
    "\n",
    "          if (is.null(window_right)){\n",
    "            window_right<- (max(as.numeric(peakm[baitID == id][baitChr == oeChr][delange_susie == TRUE]$dist)))/1000 + 20\n",
    "          }else{\n",
    "            window_right = window_right\n",
    "          }\n",
    "        }else{\n",
    "          if (is.null(window_left)){\n",
    "            window_left<- (min(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist)))/1000 - 20\n",
    "          }else{\n",
    "            window_left = window_left\n",
    "          }\n",
    "\n",
    "          if (is.null(window_right)){\n",
    "            window_right<- (max(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist)))/1000 + 20\n",
    "          }else{\n",
    "            window_right = window_right\n",
    "          }\n",
    "        }\n",
    "        \n",
    "        midpoint <- (bmap[V4==id]$V2+bmap[V4==id]$V3)/2000 # also convert to kb scale by dividing by 1000\n",
    "\n",
    "        if(window_left<0){\n",
    "          bigStart <- midpoint-abs(window_left)\n",
    "        }else{\n",
    "          bigStart <- midpoint-10\n",
    "        }\n",
    "\n",
    "        if(window_right>=0){\n",
    "          bigEnd <- midpoint + abs(window_right)\n",
    "        }else{\n",
    "          bigEnd <- midpoint+10\n",
    "        }\n",
    "        \n",
    "        chrom <- bmap[V4==id][1,]$V1\n",
    "        ensg_id <- delange_susie_res[gene == gene_name]$ensg\n",
    "\n",
    "        peakm_sel <- peakm[baitChr == chrom][((baitStart > bigStart*1000) & (oeEnd < bigEnd*1000)) | \n",
    "                                         ((oeStart > bigStart*1000) & (baitEnd < bigEnd*1000))]\n",
    "\n",
    "        \n",
    "        current_fn <- paste0(\"~/miniPCHiC/hILCs/hILCs_run2/git_ILCs_LOLA/LOLA_working_files/\", \n",
    "                             gene_name, \"_\",id, \"_\", window_left, \"kb_left_\", window_right, \"kb_right_zoom_\", zoom, \"all_ints_\", all_ints, \"_deLange_chic_abc_fres_epi_sushi.pdf\")\n",
    "        \n",
    "        height <- 20\n",
    "        npanels <- 13\n",
    "        pdf(\n",
    "          current_fn, \n",
    "          height = height, \n",
    "          width = height * 0.7070707070707# A4 proportions\n",
    "        ) \n",
    "        layout(\n",
    "          mat = matrix(c(1 : npanels), byrow = TRUE, nrow = npanels, ncol = 1), \n",
    "          heights = c(2, 1,1,1,1,1,1,1,1,1, 3, 3, 4)\n",
    "        )\n",
    "        par(mar = c(2, 4.1, 2, 4.1))\n",
    "        \n",
    "\n",
    "        #------------------------------------ Plot gene-track\n",
    "        message(\"Plotting gene track\")\n",
    "\n",
    "        # Use plotGenes to plot the gene track\n",
    "        \n",
    "        gene_sel <- gene_track_plot(gene_name = gene_name, chrom = chrom, bmap = bmap, bigStart = bigStart, bigEnd = bigEnd, midpoint = midpoint)\n",
    "        \n",
    "        pgr <- plotGenes(\n",
    "            gene_sel[, .(chrom, start, stop, gene, score = \".\", strand)], \n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            col = gene_sel$colour,\n",
    "            bheight = 0.3, \n",
    "            lheight = 0.05,\n",
    "            maxrows = 10000,\n",
    "        #     types = \"exon\",\n",
    "            plotgenetype = \"arrow\",\n",
    "            wigglefactor = 0.05,\n",
    "            labelat = \"middle\",\n",
    "            #fontsize = 2\n",
    "        )\n",
    "        \n",
    "        # Plot genome coordinate bar\n",
    "        labelgenome(\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          n = 10,\n",
    "          scale = \"Kb\"\n",
    "        )\n",
    "        \n",
    "         \n",
    "        #------------------------------------\n",
    "        # Plot ChIP-Seq pileups\n",
    "        message(\"Plotting pileups\")\n",
    "\n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(\n",
    "            beddata = atac,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            main = \"ATAC\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(\n",
    "            beddata = k27ac,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"H3K27ac\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        if (nrow(k36me3[peak_start>bigStart & peak_end<bigEnd])>0) {\n",
    "            \n",
    "            plotBed(\n",
    "                beddata = k36me3,\n",
    "                chrom = chrom,\n",
    "                chromstart = bigStart*1000,\n",
    "                chromend = bigEnd*1000,\n",
    "                color = \"grey16\",\n",
    "                ylab = \"Read count\",\n",
    "                rowlabels = \"H3K36me3\",\n",
    "                rowlabelcol= \"grey16\",\n",
    "                wiggle=0.0001,\n",
    "                rowlabelcex = 2\n",
    "        )\n",
    "\n",
    "        }else { # If there are no significant interactions, plot an empty frame\n",
    "              plotBedpe(\n",
    "                bedpedata = data.frame(), \n",
    "                chrom = chrom, \n",
    "                chromstart = bigStart*1000, \n",
    "                chromend = bigEnd*1000, \n",
    "                bty = \"o\"\n",
    "              )\n",
    "        }\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = ctcf,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"CTCF\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = sa1,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"SA1\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = ikzf1,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"IKZF1\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = batf,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"BATF\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = nfkb,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"NFKB\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = atf2,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"ATF2\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "\n",
    "        #------------------------------------\n",
    "        # Plot SNP posterior probabilities that intersect the plotting window\n",
    "\n",
    "        message(\"Plotting de Lange SNP posteriors\")\n",
    "        ppi_deLange_sel <- ppi_deLange[chr ==chrom][(pos > bigStart*1000) & (pos < bigEnd*1000)]\n",
    "        plot(\n",
    "            x = ppi_deLange_sel[, pos], \n",
    "            y = ppi_deLange_sel[, ppi],\n",
    "            cex = 2,\n",
    "            xaxs = \"i\",\n",
    "            xlim = c(bigStart*1000, bigEnd*1000),\n",
    "            ylim = c(0, max(ppi_deLange_sel$ppi)),\n",
    "            xlab = NA,\n",
    "            ylab = \"Posterior probability of variant being causal\",\n",
    "            col = adjustcolor(\"black\", alpha.f = 0.9)\n",
    "        )\n",
    "\n",
    "        #------------------------------------\n",
    "        # Plot SNP posterior probabilities that intersect the plotting window\n",
    "        message(\"Plotting de Lange SuSiE SNP posteriors\")\n",
    "        ppi_deLange_susie_sel <- ppi_deLange_susie[chr ==chrom][(pos > bigStart*1000) & (pos < bigEnd*1000)]\n",
    "        plot(\n",
    "            x = ppi_deLange_susie_sel[, pos],\n",
    "            y = ppi_deLange_susie_sel[, ppi],\n",
    "            xlim = c(bigStart*1000, bigEnd*1000),\n",
    "            ylim = c(0, max(ppi_deLange_susie_sel$ppi)),\n",
    "            cex = 2, \n",
    "            xaxs = \"i\",\n",
    "            xlab = NA,\n",
    "            ylab = \"Posterior probability of variant being causal\",\n",
    "            col = adjustcolor(\"black\", alpha.f = 0.9)\n",
    "        )\n",
    "\n",
    "       \n",
    "        #------------------------------------\n",
    "        # Plot Promoter interactions that intersect the plotting window\n",
    "        \n",
    "        message(\"Plotting PCHiC arcs\")\n",
    "        \n",
    "        \n",
    "        if(nrow(peakm_sel[baitName == gene_name]) > 0){\n",
    "\n",
    "            if (all_ints == TRUE){\n",
    "                plotBedpe(\n",
    "                    bedpedata = peakm_sel, \n",
    "                    chrom = chrom, \n",
    "                    chromstart = bigStart*1000, \n",
    "                    chromend = bigEnd*1000,\n",
    "                    color = peakm_sel$colour,\n",
    "                    heights = peakm_sel$score, \n",
    "                    flip = TRUE, \n",
    "                    plottype = \"loops\", \n",
    "                    bty = \"o\",\n",
    "                    main = \"promoter interactions in ILC3a\"\n",
    "              )\n",
    "            }else{\n",
    "\n",
    "               plotBedpe(\n",
    "                    bedpedata = peakm_sel[baitName == gene_name], \n",
    "                    chrom = chrom, \n",
    "                    chromstart = bigStart*1000, \n",
    "                    chromend = bigEnd*1000,\n",
    "                    color = peakm_sel[baitName == gene_name]$colour,\n",
    "                    heights = peakm_sel[baitName == gene_name]$score, \n",
    "                    flip = TRUE, \n",
    "                    plottype = \"loops\", \n",
    "                    bty = \"o\",\n",
    "                    main = \"promoter interactions in ILC3s\"\n",
    "                   )\n",
    "            }\n",
    "            \n",
    "        } else { # If there are no significant interactions, plot an empty frame\n",
    "              plotBedpe(\n",
    "                bedpedata = data.frame(), \n",
    "                chrom = chrom, \n",
    "                chromstart = bigStart, \n",
    "                chromend = bigEnd, \n",
    "                bty = \"o\"\n",
    "              )\n",
    "        }        \n",
    "        \n",
    "        # Plot genome coordinate bar\n",
    "        labelgenome(\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          n = 10,\n",
    "          scale = \"Kb\"\n",
    "        )\n",
    "        \n",
    "        #------------------------------------ Highlighting baits\n",
    "        message(\"Highlighting baits\")\n",
    "        \n",
    "        par(mfrow = c(1, 1))\n",
    "        bmap_sel <- bmap[V1 == chrom][V2>bigStart*1000 & V3<bigEnd*1000]\n",
    "        rect(\n",
    "          xleft = bmap_sel[, V2], \n",
    "          xright = bmap_sel[, V3],\n",
    "          ybottom = -50, \n",
    "          ytop = 20, \n",
    "          col = adjustcolor(\"steelblue\", alpha.f = 0.3),\n",
    "          border = NA\n",
    "          )\n",
    "        \n",
    "        #----------------------------------\n",
    "        # Add vertical bar over the PIRs with high PPis region\n",
    "        message(\"Highlighting PIRs with PPis\")\n",
    "        \n",
    "        par(mfrow = c(1, 1))\n",
    "        rect(\n",
    "          xleft = peakm_sel[baitName == gene_name][delange_susie == TRUE][, oeStart], \n",
    "          xright = peakm_sel[baitName == gene_name][delange_susie == TRUE][, oeEnd],\n",
    "          ybottom = -50, \n",
    "          ytop = 20, \n",
    "          col = adjustcolor(\"tomato3\", alpha.f = 0.3),\n",
    "          border = NA\n",
    "          )\n",
    "\n",
    "        \n",
    "        dev.off()\n",
    "        \n",
    "    }\n",
    "        window_left <- NULL\n",
    "        window_right <- NULL\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting gene track\n",
      "\n",
      "Cache found\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting pileups\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n",
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting de Lange SNP posteriors\n",
      "\n",
      "Plotting de Lange SuSiE SNP posteriors\n",
      "\n",
      "Plotting PCHiC arcs\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Highlighting baits\n",
      "\n",
      "Highlighting PIRs with PPis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressWarnings(batch_epi_plot_short(peakm_remerge, 2314200, bmap, window_left = -30, window_right = 30, zoom = TRUE, all_ints = TRUE))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting gene track\n",
      "\n",
      "Cache found\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting pileups\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n",
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting de Lange SNP posteriors\n",
      "\n",
      "Plotting de Lange SuSiE SNP posteriors\n",
      "\n",
      "Plotting PCHiC arcs\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Highlighting baits\n",
      "\n",
      "Highlighting PIRs with PPis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressWarnings(batch_epi_plot_short(peakm_remerge, 5834362, bmap, window_left = -100, window_right = 20, zoom = TRUE, all_ints = TRUE))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting gene track\n",
      "\n",
      "Cache found\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting pileups\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n",
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting de Lange SNP posteriors\n",
      "\n",
      "Plotting de Lange SuSiE SNP posteriors\n",
      "\n",
      "Plotting PCHiC arcs\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Highlighting baits\n",
      "\n",
      "Highlighting PIRs with PPis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressWarnings(batch_epi_plot_short(peakm_remerge, 3270636, bmap, window_left = -150, window_right = 10, zoom = TRUE, all_ints = FALSE))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl <- fread(\"~/miniPCHiC/hILCs/hILCs_run2/eQTL_data/Lepik_2017_ge_blood.all.tsv.gz\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.table: 6 × 19</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>molecular_trait_id</th><th scope=col>chromosome</th><th scope=col>position</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>variant</th><th scope=col>ma_samples</th><th scope=col>maf</th><th scope=col>pvalue</th><th scope=col>beta</th><th scope=col>se</th><th scope=col>type</th><th scope=col>ac</th><th scope=col>an</th><th scope=col>r2</th><th scope=col>molecular_trait_object_id</th><th scope=col>gene_id</th><th scope=col>median_tpm</th><th scope=col>rsid</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000187583</td><td>1</td><td>10492</td><td>C</td><td>T</td><td>chr1_10492_C_T</td><td>73</td><td>0.0774947</td><td>0.7272200</td><td> 0.0418361</td><td>0.1198600</td><td>SNP</td><td>73</td><td>942</td><td>NA</td><td>ENSG00000187583</td><td>ENSG00000187583</td><td>  0.925</td><td>rs55998931</td></tr>\n",
       "\t<tr><td>ENSG00000187608</td><td>1</td><td>10492</td><td>C</td><td>T</td><td>chr1_10492_C_T</td><td>73</td><td>0.0774947</td><td>0.0143885</td><td>-0.2756400</td><td>0.1121940</td><td>SNP</td><td>73</td><td>942</td><td>NA</td><td>ENSG00000187608</td><td>ENSG00000187608</td><td>191.429</td><td>rs55998931</td></tr>\n",
       "\t<tr><td>ENSG00000187961</td><td>1</td><td>10492</td><td>C</td><td>T</td><td>chr1_10492_C_T</td><td>73</td><td>0.0774947</td><td>0.2525570</td><td> 0.0914901</td><td>0.0798615</td><td>SNP</td><td>73</td><td>942</td><td>NA</td><td>ENSG00000187961</td><td>ENSG00000187961</td><td> 10.698</td><td>rs55998931</td></tr>\n",
       "\t<tr><td>ENSG00000188290</td><td>1</td><td>10492</td><td>C</td><td>T</td><td>chr1_10492_C_T</td><td>73</td><td>0.0774947</td><td>0.4037420</td><td>-0.1028980</td><td>0.1231230</td><td>SNP</td><td>73</td><td>942</td><td>NA</td><td>ENSG00000188290</td><td>ENSG00000188290</td><td>  8.078</td><td>rs55998931</td></tr>\n",
       "\t<tr><td>ENSG00000188976</td><td>1</td><td>10492</td><td>C</td><td>T</td><td>chr1_10492_C_T</td><td>73</td><td>0.0774947</td><td>0.8356740</td><td>-0.0129690</td><td>0.0624863</td><td>SNP</td><td>73</td><td>942</td><td>NA</td><td>ENSG00000188976</td><td>ENSG00000188976</td><td> 30.355</td><td>rs55998931</td></tr>\n",
       "\t<tr><td>ENSG00000225880</td><td>1</td><td>10492</td><td>C</td><td>T</td><td>chr1_10492_C_T</td><td>73</td><td>0.0774947</td><td>0.2337480</td><td> 0.1343220</td><td>0.1126540</td><td>SNP</td><td>73</td><td>942</td><td>NA</td><td>ENSG00000225880</td><td>ENSG00000225880</td><td>  2.120</td><td>rs55998931</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 19\n",
       "\\begin{tabular}{lllllllllllllllllll}\n",
       " molecular\\_trait\\_id & chromosome & position & ref & alt & variant & ma\\_samples & maf & pvalue & beta & se & type & ac & an & r2 & molecular\\_trait\\_object\\_id & gene\\_id & median\\_tpm & rsid\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <int> & <int> & <lgl> & <chr> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000187583 & 1 & 10492 & C & T & chr1\\_10492\\_C\\_T & 73 & 0.0774947 & 0.7272200 &  0.0418361 & 0.1198600 & SNP & 73 & 942 & NA & ENSG00000187583 & ENSG00000187583 &   0.925 & rs55998931\\\\\n",
       "\t ENSG00000187608 & 1 & 10492 & C & T & chr1\\_10492\\_C\\_T & 73 & 0.0774947 & 0.0143885 & -0.2756400 & 0.1121940 & SNP & 73 & 942 & NA & ENSG00000187608 & ENSG00000187608 & 191.429 & rs55998931\\\\\n",
       "\t ENSG00000187961 & 1 & 10492 & C & T & chr1\\_10492\\_C\\_T & 73 & 0.0774947 & 0.2525570 &  0.0914901 & 0.0798615 & SNP & 73 & 942 & NA & ENSG00000187961 & ENSG00000187961 &  10.698 & rs55998931\\\\\n",
       "\t ENSG00000188290 & 1 & 10492 & C & T & chr1\\_10492\\_C\\_T & 73 & 0.0774947 & 0.4037420 & -0.1028980 & 0.1231230 & SNP & 73 & 942 & NA & ENSG00000188290 & ENSG00000188290 &   8.078 & rs55998931\\\\\n",
       "\t ENSG00000188976 & 1 & 10492 & C & T & chr1\\_10492\\_C\\_T & 73 & 0.0774947 & 0.8356740 & -0.0129690 & 0.0624863 & SNP & 73 & 942 & NA & ENSG00000188976 & ENSG00000188976 &  30.355 & rs55998931\\\\\n",
       "\t ENSG00000225880 & 1 & 10492 & C & T & chr1\\_10492\\_C\\_T & 73 & 0.0774947 & 0.2337480 &  0.1343220 & 0.1126540 & SNP & 73 & 942 & NA & ENSG00000225880 & ENSG00000225880 &   2.120 & rs55998931\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 19\n",
       "\n",
       "| molecular_trait_id &lt;chr&gt; | chromosome &lt;int&gt; | position &lt;int&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | variant &lt;chr&gt; | ma_samples &lt;int&gt; | maf &lt;dbl&gt; | pvalue &lt;dbl&gt; | beta &lt;dbl&gt; | se &lt;dbl&gt; | type &lt;chr&gt; | ac &lt;int&gt; | an &lt;int&gt; | r2 &lt;lgl&gt; | molecular_trait_object_id &lt;chr&gt; | gene_id &lt;chr&gt; | median_tpm &lt;dbl&gt; | rsid &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000187583 | 1 | 10492 | C | T | chr1_10492_C_T | 73 | 0.0774947 | 0.7272200 |  0.0418361 | 0.1198600 | SNP | 73 | 942 | NA | ENSG00000187583 | ENSG00000187583 |   0.925 | rs55998931 |\n",
       "| ENSG00000187608 | 1 | 10492 | C | T | chr1_10492_C_T | 73 | 0.0774947 | 0.0143885 | -0.2756400 | 0.1121940 | SNP | 73 | 942 | NA | ENSG00000187608 | ENSG00000187608 | 191.429 | rs55998931 |\n",
       "| ENSG00000187961 | 1 | 10492 | C | T | chr1_10492_C_T | 73 | 0.0774947 | 0.2525570 |  0.0914901 | 0.0798615 | SNP | 73 | 942 | NA | ENSG00000187961 | ENSG00000187961 |  10.698 | rs55998931 |\n",
       "| ENSG00000188290 | 1 | 10492 | C | T | chr1_10492_C_T | 73 | 0.0774947 | 0.4037420 | -0.1028980 | 0.1231230 | SNP | 73 | 942 | NA | ENSG00000188290 | ENSG00000188290 |   8.078 | rs55998931 |\n",
       "| ENSG00000188976 | 1 | 10492 | C | T | chr1_10492_C_T | 73 | 0.0774947 | 0.8356740 | -0.0129690 | 0.0624863 | SNP | 73 | 942 | NA | ENSG00000188976 | ENSG00000188976 |  30.355 | rs55998931 |\n",
       "| ENSG00000225880 | 1 | 10492 | C | T | chr1_10492_C_T | 73 | 0.0774947 | 0.2337480 |  0.1343220 | 0.1126540 | SNP | 73 | 942 | NA | ENSG00000225880 | ENSG00000225880 |   2.120 | rs55998931 |\n",
       "\n"
      ],
      "text/plain": [
       "  molecular_trait_id chromosome position ref alt variant        ma_samples\n",
       "1 ENSG00000187583    1          10492    C   T   chr1_10492_C_T 73        \n",
       "2 ENSG00000187608    1          10492    C   T   chr1_10492_C_T 73        \n",
       "3 ENSG00000187961    1          10492    C   T   chr1_10492_C_T 73        \n",
       "4 ENSG00000188290    1          10492    C   T   chr1_10492_C_T 73        \n",
       "5 ENSG00000188976    1          10492    C   T   chr1_10492_C_T 73        \n",
       "6 ENSG00000225880    1          10492    C   T   chr1_10492_C_T 73        \n",
       "  maf       pvalue    beta       se        type ac an  r2\n",
       "1 0.0774947 0.7272200  0.0418361 0.1198600 SNP  73 942 NA\n",
       "2 0.0774947 0.0143885 -0.2756400 0.1121940 SNP  73 942 NA\n",
       "3 0.0774947 0.2525570  0.0914901 0.0798615 SNP  73 942 NA\n",
       "4 0.0774947 0.4037420 -0.1028980 0.1231230 SNP  73 942 NA\n",
       "5 0.0774947 0.8356740 -0.0129690 0.0624863 SNP  73 942 NA\n",
       "6 0.0774947 0.2337480  0.1343220 0.1126540 SNP  73 942 NA\n",
       "  molecular_trait_object_id gene_id         median_tpm rsid      \n",
       "1 ENSG00000187583           ENSG00000187583   0.925    rs55998931\n",
       "2 ENSG00000187608           ENSG00000187608 191.429    rs55998931\n",
       "3 ENSG00000187961           ENSG00000187961  10.698    rs55998931\n",
       "4 ENSG00000188290           ENSG00000188290   8.078    rs55998931\n",
       "5 ENSG00000188976           ENSG00000188976  30.355    rs55998931\n",
       "6 ENSG00000225880           ENSG00000225880   2.120    rs55998931"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(eqtl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "batch_epi_plot_short <- function(peakm, bait_list, bmap, window_left = window_left, window_right = window_right, zoom = zoom, all_ints = all_ints, idonly = TRUE){\n",
    "\n",
    "    ## e.g. peakm can be ilc_chic_abc_peakm\n",
    "    for (id in bait_list){\n",
    "        \n",
    "    gene_name <- bmap[V4 == id]$V5[1]\n",
    "\n",
    "        if(zoom == TRUE){\n",
    "\n",
    "          if (is.null(window_left)){\n",
    "            window_left<- (min(as.numeric(peakm[baitID == id][baitChr == oeChr][delange_susie == TRUE]$dist)))/1000 - 20\n",
    "          }else{\n",
    "            window_left = window_left\n",
    "          }\n",
    "\n",
    "          if (is.null(window_right)){\n",
    "            window_right<- (max(as.numeric(peakm[baitID == id][baitChr == oeChr][delange_susie == TRUE]$dist)))/1000 + 20\n",
    "          }else{\n",
    "            window_right = window_right\n",
    "          }\n",
    "        }else{\n",
    "          if (is.null(window_left)){\n",
    "            window_left<- (min(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist)))/1000 - 20\n",
    "          }else{\n",
    "            window_left = window_left\n",
    "          }\n",
    "\n",
    "          if (is.null(window_right)){\n",
    "            window_right<- (max(as.numeric(peakm[baitID == id][baitChr == oeChr]$dist)))/1000 + 20\n",
    "          }else{\n",
    "            window_right = window_right\n",
    "          }\n",
    "        }\n",
    "        \n",
    "        midpoint <- (bmap[V4==id]$V2+bmap[V4==id]$V3)/2000 # also convert to kb scale by dividing by 1000\n",
    "\n",
    "        if(window_left<0){\n",
    "          bigStart <- midpoint-abs(window_left)\n",
    "        }else{\n",
    "          bigStart <- midpoint-10\n",
    "        }\n",
    "\n",
    "        if(window_right>=0){\n",
    "          bigEnd <- midpoint + abs(window_right)\n",
    "        }else{\n",
    "          bigEnd <- midpoint+10\n",
    "        }\n",
    "        \n",
    "        chrom <- bmap[V4==id][1,]$V1\n",
    "        ensg_id <- delange_susie_res[gene == gene_name]$ensg\n",
    "\n",
    "        peakm_sel <- peakm[baitChr == chrom][((baitStart > bigStart*1000) & (oeEnd < bigEnd*1000)) | \n",
    "                                         ((oeStart > bigStart*1000) & (baitEnd < bigEnd*1000))]\n",
    "\n",
    "        eqtl_sel_CLN3 <- eqtl[gene_id == ensg_id][chromosome == chrom][position/1000> bigStart & position/1000<bigEnd]\n",
    " \n",
    "        \n",
    "        current_fn <- paste0(\"~/miniPCHiC/hILCs/hILCs_run2/git_ILCs_LOLA/LOLA_working_files/\", \n",
    "                             gene_name, \"_\",id, \"_\", window_left, \"kb_left_\", window_right, \"kb_right_zoom_\", zoom, \"all_ints_\", all_ints, \"_deLange_chic_abc_fres_epi_sushi.pdf\")\n",
    "        \n",
    "        height <- 20\n",
    "        npanels <- 14\n",
    "        pdf(\n",
    "          current_fn, \n",
    "          height = height, \n",
    "          width = height * 0.7070707070707# A4 proportions\n",
    "        ) \n",
    "        layout(\n",
    "          mat = matrix(c(1 : npanels), byrow = TRUE, nrow = npanels, ncol = 1), \n",
    "          heights = c(2, 1,1,1,1,1,1,1,1,1, 3, 3, 3, 3, 4)\n",
    "        )\n",
    "        par(mar = c(2, 4.1, 2, 4.1))\n",
    "        \n",
    "\n",
    "        #------------------------------------ Plot gene-track\n",
    "        message(\"Plotting gene track\")\n",
    "\n",
    "        # Use plotGenes to plot the gene track\n",
    "        \n",
    "        gene_sel <- gene_track_plot(gene_name = gene_name, chrom = chrom, bmap = bmap, bigStart = bigStart, bigEnd = bigEnd, midpoint = midpoint)\n",
    "        \n",
    "        pgr <- plotGenes(\n",
    "            gene_sel[, .(chrom, start, stop, gene, score = \".\", strand)], \n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            col = gene_sel$colour,\n",
    "            bheight = 0.3, \n",
    "            lheight = 0.05,\n",
    "            maxrows = 10000,\n",
    "        #     types = \"exon\",\n",
    "            plotgenetype = \"arrow\",\n",
    "            wigglefactor = 0.05,\n",
    "            labelat = \"middle\",\n",
    "            #fontsize = 2\n",
    "        )\n",
    "        \n",
    "        # Plot genome coordinate bar\n",
    "        labelgenome(\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          n = 10,\n",
    "          scale = \"Kb\"\n",
    "        )\n",
    "        \n",
    "         \n",
    "        #------------------------------------\n",
    "        # Plot ChIP-Seq pileups\n",
    "        message(\"Plotting pileups\")\n",
    "\n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(\n",
    "            beddata = atac,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"ATAC\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(\n",
    "            beddata = k27ac,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"H3K27ac\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "#         # message(paste0(\"Plotting pileup\"))\n",
    "#         if (nrow(k36me3[chr == chrom][peak_start>bigStart & peak_end<bigEnd])>0) {\n",
    "            \n",
    "#             plotBed(\n",
    "#                 beddata = k36me3,\n",
    "#                 chrom = chrom,\n",
    "#                 chromstart = bigStart*1000,\n",
    "#                 chromend = bigEnd*1000,\n",
    "#                 color = \"grey16\",\n",
    "#                 ylab = \"Read count\",\n",
    "#                 rowlabels = \"H3K36me3\",\n",
    "#                 rowlabelcol= \"grey16\",\n",
    "#                 wiggle=0.0001,\n",
    "#                 rowlabelcex = 2\n",
    "#                 )\n",
    "\n",
    "#         }else { # If there are no significant interactions, plot an empty frame\n",
    "#               plotBedpe(\n",
    "#                 bedpedata = data.frame(), \n",
    "#                 chrom = chrom, \n",
    "#                 chromstart = bigStart*1000, \n",
    "#                 chromend = bigEnd*1000, \n",
    "#                 bty = \"o\"\n",
    "#               )\n",
    "#         }\n",
    "        \n",
    "        plotBed(beddata = k36me3,\n",
    "                chrom = chrom,\n",
    "                chromstart = bigStart*1000,\n",
    "                chromend = bigEnd*1000,\n",
    "                color = \"grey16\",\n",
    "                ylab = \"Read count\",\n",
    "                rowlabels = \"H3K36me3\",\n",
    "                rowlabelcol= \"grey16\",\n",
    "                wiggle=0.0001,\n",
    "                rowlabelcex = 2\n",
    "                )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = ctcf,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"CTCF\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = sa1,\n",
    "            chrom = chrom,\n",
    "            chromstart = bigStart*1000,\n",
    "            chromend = bigEnd*1000,\n",
    "            color = \"grey16\",\n",
    "            ylab = \"Read count\",\n",
    "            rowlabels = \"SA1\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = ikzf1,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"IKZF1\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = batf,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"BATF\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = nfkb,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"NFKB\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "        \n",
    "        # message(paste0(\"Plotting pileup\"))\n",
    "        plotBed(beddata = atf2,\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          color = \"grey16\",\n",
    "          ylab = \"Read count\",\n",
    "            rowlabels = \"ATF2\",\n",
    "            rowlabelcol= \"grey16\",\n",
    "            wiggle=0.0001,\n",
    "            rowlabelcex = 2\n",
    "        )\n",
    "\n",
    "        message(\"Plotting eQTL data\")\n",
    "        \n",
    "        plot(\n",
    "            x = eqtl_sel_CLN3[chromosome == chrom][, position], \n",
    "            y = eqtl_sel_CLN3[chromosome == chrom][, (-1)*log(pvalue, 10)],\n",
    "            cex = 2,\n",
    "            xaxs = \"i\",\n",
    "            xlim = c(bigStart*1000, bigEnd*1000),\n",
    "            ylim = c(0, 13),\n",
    "            xlab = NA,\n",
    "            ylab = \"eQTL for CLN3 in blood (Lepik et al., 2017)\",\n",
    "            col = adjustcolor(\"black\", alpha.f = 0.9)\n",
    "        )\n",
    "        \n",
    "        #------------------------------------\n",
    "        # Plot SNP posterior probabilities that intersect the plotting window\n",
    "        message(\"Plotting raw CD data posteriors\")\n",
    "        \n",
    "        plot(\n",
    "            x = raw_deLange_data[hm_chrom == chrom][, hm_pos], \n",
    "            y = raw_deLange_data[hm_chrom == chrom][, (-1)*log(p_value, 10)],\n",
    "            cex = 2,\n",
    "            xaxs = \"i\",\n",
    "            xlim = c(bigStart*1000, bigEnd*1000),\n",
    "            ylim = c(0, max((-1)*log(raw_deLange_data[hm_chrom == chrom][(hm_pos > bigStart*1000) & (hm_pos < bigEnd*1000)]$p_value, 10))),\n",
    "            xlab = NA,\n",
    "            ylab = \"CD GWAS -log10(p-value)\",\n",
    "            col = adjustcolor(\"black\", alpha.f = 0.9)\n",
    "        )\n",
    "\n",
    "        #------------------------------------\n",
    "        # Plot SNP posterior probabilities that intersect the plotting window\n",
    "        message(\"Plotting de Lange SuSiE SNP posteriors\")\n",
    "        ppi_deLange_susie_sel <- ppi_deLange_susie[chr ==chrom][(pos > bigStart*1000) & (pos < bigEnd*1000)]\n",
    "        plot(\n",
    "            x = ppi_deLange_susie_sel[chr ==chrom][, pos],\n",
    "            y = ppi_deLange_susie_sel[chr ==chrom][, ppi],\n",
    "            xlim = c(bigStart*1000, bigEnd*1000),\n",
    "            ylim = c(0, max(ppi_deLange_susie_sel[chr ==chrom]$ppi)),\n",
    "            cex = 2, \n",
    "            xaxs = \"i\",\n",
    "            xlab = NA,\n",
    "            ylab = \"Posterior probability of variant being causal\",\n",
    "            col = adjustcolor(\"black\", alpha.f = 0.9)\n",
    "        )\n",
    "\n",
    "        \n",
    "\n",
    "        #------------------------------------\n",
    "        # Plot Promoter interactions that intersect the plotting window\n",
    "        \n",
    "        message(\"Plotting PCHiC arcs\")\n",
    "        \n",
    "        \n",
    "        if(nrow(peakm_sel[baitName == gene_name]) > 0){\n",
    "\n",
    "            if (all_ints == TRUE){\n",
    "                plotBedpe(\n",
    "                    bedpedata = peakm_sel, \n",
    "                    chrom = chrom, \n",
    "                    chromstart = bigStart*1000, \n",
    "                    chromend = bigEnd*1000,\n",
    "                    color = peakm_sel$colour,\n",
    "                    heights = peakm_sel$score, \n",
    "                    flip = TRUE, \n",
    "                    plottype = \"loops\", \n",
    "                    bty = \"o\",\n",
    "                    main = \"promoter interactions in ILC3a\"\n",
    "              )\n",
    "            }else{\n",
    "\n",
    "               plotBedpe(\n",
    "                    bedpedata = peakm_sel[baitName == gene_name], \n",
    "                    chrom = chrom, \n",
    "                    chromstart = bigStart*1000, \n",
    "                    chromend = bigEnd*1000,\n",
    "                    color = peakm_sel[baitName == gene_name]$colour,\n",
    "                    heights = peakm_sel[baitName == gene_name]$score, \n",
    "                    flip = TRUE, \n",
    "                    plottype = \"loops\", \n",
    "                    bty = \"o\",\n",
    "                    main = \"promoter interactions in ILC3s\"\n",
    "                   )\n",
    "            }\n",
    "            \n",
    "        } else { # If there are no significant interactions, plot an empty frame\n",
    "              plotBedpe(\n",
    "                bedpedata = data.frame(), \n",
    "                chrom = chrom, \n",
    "                chromstart = bigStart, \n",
    "                chromend = bigEnd, \n",
    "                bty = \"o\"\n",
    "              )\n",
    "        }        \n",
    "        \n",
    "        # Plot genome coordinate bar\n",
    "        labelgenome(\n",
    "          chrom = chrom,\n",
    "          chromstart = bigStart*1000,\n",
    "          chromend = bigEnd*1000,\n",
    "          n = 10,\n",
    "          scale = \"Kb\"\n",
    "        )\n",
    "        \n",
    "        #------------------------------------ Highlighting baits\n",
    "        message(\"Highlighting baits\")\n",
    "        \n",
    "        par(mfrow = c(1, 1))\n",
    "        bmap_sel <- bmap[V1 == chrom][V2>bigStart*1000 & V3<bigEnd*1000]\n",
    "        rect(\n",
    "          xleft = bmap_sel[, V2], \n",
    "          xright = bmap_sel[, V3],\n",
    "          ybottom = -50, \n",
    "          ytop = 20, \n",
    "          col = adjustcolor(\"steelblue\", alpha.f = 0.3),\n",
    "          border = NA\n",
    "          )\n",
    "        \n",
    "        #----------------------------------\n",
    "        # Add vertical bar over the PIRs with high PPis region\n",
    "        message(\"Highlighting PIRs with PPis\")\n",
    "        \n",
    "        par(mfrow = c(1, 1))\n",
    "        rect(\n",
    "          xleft = peakm_sel[baitName == gene_name][delange_susie == TRUE][, oeStart], \n",
    "          xright = peakm_sel[baitName == gene_name][delange_susie == TRUE][, oeEnd],\n",
    "          ybottom = -50, \n",
    "          ytop = 20, \n",
    "          col = adjustcolor(\"tomato3\", alpha.f = 0.3),\n",
    "          border = NA\n",
    "          )\n",
    "\n",
    "        \n",
    "        dev.off()\n",
    "        \n",
    "    }\n",
    "        window_left <- NULL\n",
    "        window_right <- NULL\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting gene track\n",
      "\n",
      "Cache found\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting pileups\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n",
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Plotting eQTL data\n",
      "\n",
      "Plotting raw CD data posteriors\n",
      "\n",
      "Plotting de Lange SuSiE SNP posteriors\n",
      "\n",
      "Plotting PCHiC arcs\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"yes\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Highlighting baits\n",
      "\n",
      "Highlighting PIRs with PPis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressWarnings(batch_epi_plot_short(peakm_remerge, 2314200, bmap, window_left = -30, window_right = 30, zoom = TRUE, all_ints = TRUE))\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:new-r-env]",
   "language": "R",
   "name": "conda-env-new-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
